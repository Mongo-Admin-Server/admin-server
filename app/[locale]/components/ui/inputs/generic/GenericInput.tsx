import { useState } from 'react';

import styles from './input.module.scss';

import SvgIcon from '@/app/[locale]/components/ui/icon/SvgIcon';

interface GenericInputProps {
  label: string;
  value: string;
  // eslint-disable-next-line no-unused-vars
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  type: 'text' | 'password' | 'email' | 'number';
  placeholder?: string;
  error?: string;
}

const GenericInput = ({
  label,
  value,
  onChange,
  type,
  placeholder,
  error,
}: GenericInputProps) => {
  const [showPassword, setShowPassword] = useState(false);

  return (
    <div className={styles.container}>
      <label className={styles.label}>{label}</label>
      <input
        className={`${styles.input} ${error ? styles.inputError : ''}`}
        type={type === 'password' && showPassword ? 'text' : type}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
      />

      {type === 'password' && (
        <div
          className={styles.iconContainer}
          onClick={() => setShowPassword(!showPassword)}
        >
          {showPassword ? (
            <SvgIcon icon_name="eye" />
          ) : (
            <SvgIcon icon_name="eyeSlash" />
          )}
        </div>
      )}

      {error && <p className={styles.error}>{error}</p>}
    </div>
  );
};

export default GenericInput;
