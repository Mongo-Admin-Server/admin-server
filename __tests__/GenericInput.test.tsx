/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import GenericInput from '@/app/[locale]/components/ui/inputs/generic/GenericInput';

const mock = {
  label: 'form input',
  value: 'form input value',
};

describe('GenericInput', () => {
  it('renders', () => {
    const handleChange = jest.fn();
    const { container } = render(
      <GenericInput 
        label={mock.label}
        value={mock.value}
        type='text'
        onChange={handleChange} 
      />
    );
    expect(container).toMatchSnapshot();
  })
})