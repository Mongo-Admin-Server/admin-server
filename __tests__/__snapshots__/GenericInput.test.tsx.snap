// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericInput renders 1`] = `
<div>
  <div
    class="container"
  >
    <label
      class="label"
    >
      form input
    </label>
    <input
      class="input "
      type="text"
      value="form input value"
    />
  </div>
</div>
`;
